prompting/datasets/wiki.py

```diff
@@ -209,49 +160,46 @@ def get(
         if not sections:
             return None
 
-        key = header, section_title = selector(list(sections.keys()))
+        key = header, section_title = random.choice(list(sections.keys()))
         content = "\n".join(sections[key])
         section_length = len(content.split())
-        context = {
-            "title": name,  # title of wiki article
-            "topic": header or section_title,  # title of wiki section
-            "subtopic": section_title,
-            "content": content,
-            "internal_links": list(filter(lambda x: x not in exclude, page.sections)),
-            "external_links": most_relevant_links(page, num_links=self.max_links),
-            "tags": filter_categories(page.categories, exclude=self.EXCLUDE_CATEGORIES),
-            "source": "Wikipedia",
-            "extra": {
+
+        context = Context(
+            title=name,
+            topic=header or section_title,
+            subtopic=section_title,
+            content=section_title,
+            internal_links=list(filter(lambda x: x not in exclude, page.sections)),
+            external_links=most_relevant_links(page, num_links=self.max_links),
+            tags=filter_categories(page.categories, exclude=self.EXCLUDE_CATEGORIES),
+            source="Wikipedia",
+            extra={
                 "url": page.url,
                 "page_length": len(page.content.split()),
                 "section_length": section_length,
             },
-        }
+        )
         try:
             CACHED_ARTICLES.put(context, block=False)
         except Full:
             bt.logging.debug("Cache is full. Skipping article until cache is emptied.")
         return context
 
-    def search(self, name, results=3, selector: Selector = None) -> Dict:
-        titles = _wiki_search(name, results=results)
-        title = selector(titles)
-        return self.get(title, selector=selector)
+    def search(self, name, results=3) -> Context:
+        titles = _wikipedia_search(name, results=results)
+        title = random.choice(titles)
+        return self.get(title)
 
-    def random(self, pages=10, seed=None, selector: Selector = None, **kwargs) -> Dict:
-        titles = (
-            wiki.random(pages=pages)
-            if seed is None
-            else _get_random_titles(pages=pages, seed=seed)
-        )
-        title = selector(titles)
-        return self.get(title, selector=selector)
+    def random(self, pages=10, seed=None) -> Context:
+        titles = wikipedia.random(pages=pages) if seed is None else _get_random_titles(pages=pages, seed=seed)
```

Comment:
Can we consolidate this into a single call?  `_get_random_titles` should be able to check if `seed is None`